### ISSUE TITLE
blank lines between import groups is ignored

### ISSUE THREAD

**Issue by scrocquesel** [redhat-developer/vscode-java#2980]
With a format settings file containing 

`<setting id="org.eclipse.jdt.core.formatter.blank_lines_between_import_groups" value="1"/>`

and `java.completion.importOrder` specifying groups, all imports are still not spaced.

```
"java.completion.importOrder": [
        "java",
        "javax",
        "org",
        "com",
        "",
        "#"
    ]
```

##### Environment
- Operating System: Linux
- JDK version: 
- Visual Studio Code version: 1.76
- Java extension version: 1.15.0

##### Steps To Reproduce
1. Organize imports will remove any blank lines
2. Format document do not add or remove blank lines

##### Current Result
```
import java.time.Duration;
import java.util.HashSet;
import java.util.Optional;
import java.util.Set;
import java.util.function.Consumer;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
```

##### Expected Result
```
import java.time.Duration;
import java.util.HashSet;
import java.util.Optional;
import java.util.Set;
import java.util.function.Consumer;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
```

**Comment by snjeza** [comment]
@scrocquesel I can't reproduce the issue.
Where you set `<setting id="org.eclipse.jdt.core.formatter.blank_lines_between_import_groups" value="1"/>` ?

**Comment by scrocquesel** [comment]
@snjeza In a file relative to the project. I recheck and Organize does honor the setting but not Format document. (I edit the description)

**Comment by snjeza** [comment]
@scrocquesel Could you show your user, workspace `settings.json` and `<your_project>/.settings/org.eclipse.jdt.core.prefs` ?

**Comment by scrocquesel** [comment]
> @scrocquesel Could you show your user, workspace `settings.json` and `<your_project>/.settings/org.eclipse.jdt.core.prefs` ?

Nothing java related in user settings.json and no `<your_project>/.settings/org.eclipse.jdt.core.prefs` file.
Project settings can be found at https://github.com/scrocquesel/java-operator-sdk/commit/9ed7d7d8b6c97638d27368bb224f6c5cd8e1d95b

**Comment by snjeza** [comment]
@scrocquesel Could you try right click, select  `Source Action>Organize Imports`

**Comment by scrocquesel** [comment]
Yes, this works but not format command. I understand that we need to do two command to format files according to the eclipse format settings profile.

**Comment by snjeza** [comment]
It can be reproduced in Eclipse.

**Comment by rgrunber** [comment]
Apparently this is functioning as designed :

https://bugs.eclipse.org/bugs/show_bug.cgi?id=165210#c5
https://bugs.eclipse.org/bugs/show_bug.cgi?id=572253#c1

**Comment by Nightenom** [comment]
I'm having very similar problem, however with `<setting id="org.eclipse.jdt.core.formatter.blank_lines_between_import_groups" value="-1"/>` ie. no lines between any imports. The issue is that `java.saveActions.organizeImports` runs after `editor.formatOnSave` so the formatter rules are not applied correctly
